### PR TITLE
chore: ring withdraw use dispatch precompiler

### PR DESCRIPTION
**Pull request by JayJay1024**

### CODE REVIEW

File: src/components/toolbox/withdraw/Withdraw.tsx [hunk 1]
```diff
@@ -34,10 +34,7 @@ interface WithdrawFormValues {
 }
 
 const WITHDRAW_GAS = 55000;
-const DVM_WITHDRAW_ADDRESS = '0x0000000000000000000000000000000000000015';
-const EMPTY_ADDRESS = '0x0000000000000000000000000000000000000000000000000000000000000000';
-
-const registry = new TypeRegistry();
+const DVM_DISPATCH_ADDRESS = '0x0000000000000000000000000000000000000019';
```
Comment (hackfisher) [inline src/components/toolbox/withdraw/Withdraw.tsx:38]: Add todo comment here: https://github.com/darwinia-network/darwinia-common/pull/1298

Comment (hackfisher) [reply src/components/toolbox/withdraw/Withdraw.tsx:38]: ~~Which network is supported?~~

~~Crab? or Crab and Pangolin?~~

**Comment by hackfisher** [comment]
@ZolaxZola Please help test.  

~~And I think we need to add warn in the ui for the user, about the decimal of the balance is same with Crab chain, should be 9.~~

image

**Review by hackfisher — CHANGES_REQUESTED** [review]
It seems that for here, we also support WCKTON?

Please note that CRAB and CKTON are separate and different, the new way of using dispatch is only for CRAB.

CKTON keep unchanged.

**Comment by JayJay1024** [comment]
> @ZolaxZola Please help test.
> 
> And I think we need to add warn in the ui for the user, about the decimal of the balance is same with Crab chain, should be 9.
> 
> image

Now here the precision is forced to 3.

**Comment by hackfisher** [comment]
> Now here the precision is forced to 3.

Why introduce this design?

**Review by hackfisher — DISMISSED** [review]
LGTM

**Comment by hackfisher** [comment]
> 
> ~And I think we need to add warn in the ui for the user, about the decimal of the balance is same with Crab chain, should be 9.~
> 


Revoke this suggestion, use does need to care about this.

**Comment by ZolaxZola** [comment]
> > Now here the precision is forced to 3.
> 
> Why introduce this design?

may cause this problem：cannot transfer the 0.0005 CRAB part
image

**Comment by ZolaxZola** [comment]
image
From ：0x87F71162cAAC6D9fD8f48C2342509d838578CC81

**Comment by hackfisher** [comment]
Known issues:

- Pre this pull request
  - Users are sending values to 0x19 (not destination)
  - 0x19 is unstable custom precompile
- Post this pull request
  - Users may not see values in metamask transferring, only calling to some contact 0x19 with un-readable calldata (looks dangerous)
  - In apps, the apps will require to predict the gas fee, it is hard(gas price may change in metamask), so it make the apps hard to calculate how many balances are available for transfer.  (Current solution, keep enough fees for gas cc @JayJay1024 )

**Comment by hackfisher** [comment]
@JayJay1024 @sxlwar We finally decide to use Dispatch compile, please continue with this PR, thanks.

Please take care of the consideration of UX challenges, leaving enough fees for user when the input amount, in case of confusions to users.

**Review by sxlwar — APPROVED** [review]

**Pull request merged**